CommonInternals/Hasher.cs

```diff
@@ -0,0 +1,117 @@
+/*
+ * Derived from MurmurHash2Simple,
+ * http://landman-code.blogspot.com/2009/02/c-superfasthash-and-murmurhash2.html
+ */
```

Comment:
That is the same code than the one added in nhibernate/nhibernate-core#1802. Added a common library for sharing it among cache libraries was looking as an overkill, so I have added it as a common file.